050-Kafka-Node-Pools.md

```diff
@@ -0,0 +1,806 @@
+For example:
+
+* If the `KafkaNodePool.spec.jvmOptions` does not exist but `Kafka.spec.kafka.jvmOptions` does exist, the values from `Kafka.spec.kafka.jvmOptions` will be used.
+* When `KafkaNodePool.spec.jvmOptions` exists and has `-Xmx: 1024m` and `Kafka.spec.kafka.jvmOptions` exists as well with `-Xms: 512m`, the operator will completely ignore the value from `Kafka.spec.kafka.jvmOptions` and use only the value from `KafkaNodePool.spec.jvmOptions`.
+* When `KafkaNodePool.spec.template` contains only the field `podSet.metadata.labels`, the operator will completely ignore the corresponding values from `Kafka.spec.kafka.template` (for example `podSet.metadata.annotations`, but also for example `pod.metadata.labels`).
+
+In the future, additional fields might be moved to the `KafkaNodePool` and its `.spec` section according to the needs.
+This proposal tries to minimize the effort and the scope and therefore picks up only some of the fields for the initial implementation.
+Since the existing fields in the `Kafka` CR will be used as the _defaults_, new fields can be added to `KafkaNodePool` resource in the future without any backwards compatibility concerns.
+
+### Resource naming
+
+Each node pool will be represented by its own `StrimziPodSet` and pods running the actual nodes.
+The `StrimziPodSet` will be created automatically by the operator based on the `KafkaNodePool` definition - it will not be created by the users.
+The `StrimziPodSet` will be named `<ClusterName>-<PoolName>` - e.g. `my-cluster-big-nodes`.
+The resulting pods will be named `<ClusterName>-<PoolName>-<ID>` - e.g. `my-cluster-big-nodes-5`.
+The naming of the related resources such as Services, Config Maps, or PVCs will follow the pod names.
+
+That means that when a node moves from one pool to another, all the resources will be recreated including the per-node services or storage.
+It is not possible to re-use these things not just because of the naming concerns, but also because the different pools might have different configurations and the old resources might not be re-usable.
+So a node which moves will start from _zero_ with empty disk and will need to re-sync all the data.
+
+### Matching the node pools to the Kafka cluster
+
+A particular `KafkaNodePool` resource can belong only to a single Kafka cluster.
+The node pools will use the same mechanism to define to which cluster they belong as we use today for `KafkaConnector`, `KafkaTopic` or `KafkaUSer` resources.
+`strimzi.io/cluster` label will be set on the `KafkaNodePool` and its value will be the name of the `Kafka` custom resource to which it belongs.
+For example, the label `strimzi.io/cluster: my-cluster` will mean that this node pool belongs to a Kafka cluster named `my-cluster`.
+The `Kafka` and `KafkaNodePool` resources always have to be in the same namespace.
+
+#### Example
+
+The following example shows how the `Kafka` and `KafkaNodePool` resources might combine:
```

Comment:
I would clarify "... to set up a Kafka cluster with 6 brokers in total", just to make clear to users that now they have to familiarize with not just one `replicas` field if using more node pools but taking into account the sum of them.